### PR TITLE
Mobile 3931

### PR DESCRIPTION
**Pull request by crazyserver**

### CODE REVIEW

File: src/core/features/course/services/course-helper.ts [hunk 1]
```diff
@@ -555,45 +549,6 @@ export class CoreCourseHelperProvider {
...
-     * This function is meant to be called from a context menu option. It will also modify some data like the prefetch icon.
-     *
-     * @param instance The component instance that has the context menu.
-     * @param module Module to be prefetched
-     * @param courseId Course ID the module belongs to.
-     * @param done Function to call when done. It will close the context menu.
-     * @return Promise resolved when done.
-     * @deprecated since 4.0
-     */
-    async contextMenuPrefetch(
```
Comment (dpalou) [inline src/core/features/course/services/course-helper.ts:584]: Please document the functions removed in upgrade.txt :)

File: src/core/features/grades/pages/course/course.page.ts [hunk 2]
```diff
@@ -155,6 +163,22 @@ export class CoreGradesCoursePage implements AfterViewInit, OnDestroy {
     private async fetchInitialGrades(): Promise<void> {
         try {
             await this.fetchGrades();
+
+            if (this.gradeId && this.rows) {
+                const row = this.rows.find((row) => row.id == this.gradeId);
+
+                if (row) {
+                    this.toggleRow(row, true);
+                    CoreUtils.nextTick();
```
Comment (dpalou) [inline src/core/features/grades/pages/course/course.page.ts:172]: You lack an await here. If it was working, maybe this tick isn't needed.

**Review by dpalou — CHANGES_REQUESTED** [review]

File: src/core/features/course/classes/main-resource-component.ts [hunk 3]
```diff
@@ -449,6 +447,48 @@ export class CoreCourseModuleMainResourceComponent implements OnInit, OnDestroy,
...
+                description: this.description,
+                component: this.component,
+                courseId: this.courseId,
+                hasOffline: this.hasOffline,
+            },
+        });
+
+        if (data) {
+            if (data.action == 'refresh') {
+                const modal = await CoreDomUtils.showModalLoading();
```
[the remark this answers is not in the document]
Comment (crazyserver) [reply src/core/features/course/classes/main-resource-component.ts:420]: Great improvement! Thanks!

File: src/core/features/course/services/course-helper.ts (hunk 1 above)
Comment (crazyserver) [reply src/core/features/course/services/course-helper.ts:584]: Added!

**Pull request merged**